### PR TITLE
New blog post - RIMdev Radio: Building with Astro

### PR DESCRIPTION
**Pull request by mrtrimble**
i did a thing

### CODE REVIEW

File: _posts/2022-10-12-rimdev-radio-building-with-astro.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+Astro is like [Hugo](https://gohugo.io/) and [11ty](https://11ty.dev), yet contains a few neat tricks right out of the box. The main goal of Astro is to deliver as little JavaScript to the browser as possible. It handles this by running JavaScript in the build process or when rendered server-side. That is not to say you can't run JavaScript in the browser though. In fact, you can include components from any of the major JS frameworks, such as Vue, React, and Svelte. This is possible through a concept Astro is pioneering: [component islands](https://docs.astro.build/en/concepts/islands/).
+
+I recently [delivered a talk](https://astro-friday-talk-slides.netlify.app/1) on Astro to our dev team as part of the "Friday Talks" we do here in RIMdev. I demonstrated how to build with Astro and it's server-side rendering capability. The [first demo](https://astro-friday-talk.netlify.app/) shows off it's framework agnostic abilities, and how to run JavaScript. The [second demo](http://astro-pokemon.netlify.app/) shows how you can run JavaScript server-side and deliver only HTML. These projects were fun to build, but I wanted to try and build something with a bit more substance.
```
Comment (vchinnick) [inline _posts/2022-10-12-rimdev-radio-building-with-astro.md:27]:
```suggestion
I recently [delivered a talk](https://astro-friday-talk-slides.netlify.app/1) on Astro to our dev team as part of the "Friday Talks" we do here in RIMdev. I demonstrated how to build with Astro and it's server-side rendering capability. The [first demo](https://astro-friday-talk.netlify.app/) shows off its framework agnostic abilities and how to run JavaScript. The [second demo](http://astro-pokemon.netlify.app/) shows how you can run JavaScript server-side and deliver only HTML. These projects were fun to build, but I wanted to try and build something with a bit more substance.
```

File: _posts/2022-10-12-rimdev-radio-building-with-astro.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+## Piecing it all together
+
+### Astro
+
+Everything happens inside Astro and luckily getting started is quite easy!
+
+The `npm create astro@latest` command walks you through the process of setting up a new instance. I choose the default options for everything, except opting out of using TypeScript. 
+
+From there, the new site will scaffold and there are few organizational things to take note of:
```
Comment (vchinnick) [inline _posts/2022-10-12-rimdev-radio-building-with-astro.md:58]:
```suggestion
From there, the new site will scaffold and there are a few organizational things to take note of:
```

File: _posts/2022-10-12-rimdev-radio-building-with-astro.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+- The **Pages** directory is where each site page will live. These will use the `.astro` file extension, and are capable of running JavaScript at build.  So you can generate HTML code by running certain JavaScript functionality.
+- The **Components** directory will house UI components. These can be Astro components or components for whichever framework you prefer.
+- The **Layouts** directory is where you can include components used for layout purposes.
+
+As I intended to use Sass for styling, I also added a `styles` folder to the `src` directory.
+
+Speaking of which, to setup Sass - you do need to install Sass as a dependency to use it:
```
Comment (vchinnick) [inline _posts/2022-10-12-rimdev-radio-building-with-astro.md:65]:
```suggestion
Speaking of which, to set up Sass - you do need to install Sass as a dependency to use it:
```

File: _posts/2022-10-12-rimdev-radio-building-with-astro.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+npx astro add netlify
+```
+
+Running the add Netlify integration command will automatically install necessary packages and update the `astro.config.mjs` file.
+
+Now all that is needed is to deploy the site to Netlify. I typically do this by connecting Netlify to the GitHub repo, which gives the added benefit of automatically re-deploying the site when pushing to the main branch. You can also use PR Previews to see changes to the site before a pull request is even merged to the main branch.
+
+## Final Thoughts
+
+Overall working with Astro is a super enjoyable experience. I'm looking forward to building more projects with it!
```
Comment (vchinnick) [inline _posts/2022-10-12-rimdev-radio-building-with-astro.md:221]:
```suggestion
Overall, working with Astro is a super enjoyable experience. I'm looking forward to building more projects with it!
```

**Review by vchinnick — COMMENTED** [review]
Really well done! Even as a layman, I really feel like this was straightforward and engaging. Such a cool project!

**Review by hougasian — APPROVED** [review]

**Pull request merged**